### PR TITLE
Do Post: Step 4 of week 3's walkthrough

### PR DESCRIPTION
**Pull request by freliciatucker**
- attempt at creating functional comments feature with the goal of redirecting back to /index.html url to display processed comments. 
- known bug: everytime the /index.html url is reloaded, the "fetch a greeting" button adds 1, 2, 3 to the displayed array list

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -93,16 +93,33 @@ function getGreetingWithArrow() {
   });
 }
 
+
+/**
+ * Fetches the current state of the form and builds the UI.
+ */
+function getForm() {
+    console.log("in getform;");
```
Comment (jmchristen) [inline portfolio/src/main/webapp/script.js:101]: Extraneous log statements in a few places in this file.

Comment (freliciatucker) [reply portfolio/src/main/webapp/script.js]: i deleted only this one from this pull requests. I'm working through understanding how to separate different pull requests, but i attempted to delete other log statements in different pull requests. The goal is to merge my fetch and parsing-json branches to master and then to sync the master to this branch to see if those changes were made. is this what you would suggest?

Comment (jmchristen) [reply portfolio/src/main/webapp/script.js]: I think so, yes.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -14,19 +14,95 @@
...
+    boolean iceCream = Boolean.parseBoolean(getParameter(request, "ice-cream", "false"));
+    boolean pizza = Boolean.parseBoolean(getParameter(request, "pizza", "false"));
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("name", getParameter(request, "text-input", ""));
+    commentEntity.setProperty("iceCream", iceCream);
+    commentEntity.setProperty("pizza", pizza);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+     datastore.put(commentEntity);
```
Comment (jmchristen) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:59]: I think this should be in the other, datastore PR.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -45,3 +44,74 @@ function openPictureText(num){
...
+}
+
+/**
+ * The above code is organized to show each individual step, but we can use an
+ * ES6 feature called arrow functions to shorten the code. This function
+ * combines all of the above code into a single Promise chain. You can use
+ * whichever syntax makes the most sense to you.
+ */
+function getGreetingWithArrow() {
+  fetch('/data').then(response => response.json()).then((quote) => {
```
Comment (jmchristen) [inline portfolio/src/main/webapp/script.js:86]: I get an error here when clicking the "Fetch a greeting!" button: `Unexpected number in JSON at position 1`.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -45,3 +44,74 @@ function openPictureText(num){
...
+      console.log(quote);
+    /**
+    * this line of code is commented out because I may need it later.
+    * TODO: Delete this comment or use line when final project is submitted
+    *document.getElementById('greet-container').innerText = quote;
+    */
```
Comment (jmchristen) [inline portfolio/src/main/webapp/script.js:92]: Prefer not to submit commented-out code.

**Pull request merged**